packages/flutter_tools/lib/src/drive/web_driver_service.dart

```diff
@@ -116,11 +121,16 @@ class WebDriverService extends DriverService {
       throw ToolExit('Failed to start application');
     }
 
-    _webUri = _residentRunner.uri;
-
-    if (_webUri == null) {
+    if (_residentRunner.uri == null) {
```

Comment:
Should this check move down to after processing the debugging options? We only care want to error if we couldn't produce a valid `_webUri`, right?

Reply:
I would say that both the `_webUri` and `_residentRunner.uri` should be valid. Even if you supply a different `webLaunchUrl` the original connection should still be able to be set up properly. That is why I didn't use `tryParse` for applying the `webLaunchUrl` later on if it is supplied.

Reply:
Another option would be to use `tryParse` if `webLaunchUrl` is available and throw a `ToolExit` if it returns `null`

Reply:
I don't have a strong opinion on it. It's just a nitpick, and it's fine either way, I just think it makes more sense to validate the URI we're actually going to use. But I'm fine with landing it with any of those options.